src/docs/arch-compact.dox

```diff
@@ -0,0 +1,65 @@
+/*! @arch_page arch-compact Compaction
+
+Compaction is a cooperative process between the @ref arch-btree "Btree layer" and
+the @ref arch-block "Block Manager" to reduce the on-disk footprint of WT tables. The compaction
+process is initiated by the user application by calling \c WT_SESSION::compact method. Overall, the
+compaction process involves rewriting disk blocks from the end of the on-disk file to the start of
+the file, hence giving an opportunity to reclaim the space from the end of the file through truncation
+and effectively reducing the file system reported file size. Note that the compaction is a
+"best-effort" process and does not guarantee a reduction in file size.
+
+Before understanding the compaction process, we need to understand how disk blocks are referenced by the
+checkpoints. Each checkpoint references a certain set of disk blocks for a table. When a dirty
+btree page is reconciled, a new disk block is assigned for the page and a new checkpoint starts
+referencing the new block used for the page. Once the checkpoint referencing the old block is deleted,
+the old disk block becomes available for reuse. Compaction process copies over the data from
```

Comment:
Do you think it would be more clear if we say here `Compaction process **moves** the data`?